src/py/flwr/common/config.py

```diff
@@ -215,3 +215,15 @@ def get_metadata_from_config(config: dict[str, Any]) -> tuple[str, str]:
         config["project"]["version"],
         f"{config['tool']['flwr']['app']['publisher']}/{config['project']['name']}",
     )
+
+
+def get_metadata_from_fab_filename(
+    fab_file: Union[Path, str]
+) -> tuple[str, str, str, str]:
+    """Extract metadata from the FAB filename."""
+    if isinstance(fab_file, Path):
+        fab_file_name = fab_file.stem
+    elif isinstance(fab_file, str):
+        fab_file_name = fab_file.removesuffix(".fab")
+    publisher, project_name, version, shorthash = fab_file_name.split(".")
+    return publisher, project_name, version.replace("-", "."), shorthash
```

Comment:
Maybe we should create an object for this sort of metadata, I think it would not only make the code cleaner, but also eliminate annoying order errors when dealing with such return types. But I think that would be for another PR ^^

Reply:
That's very true. I'll open another PR for this. Thanks!